hyperspy/_signals/image.py

```diff
@@ -40,3 +40,98 @@ def to_spectrum(self):
+        ----------
+        navigator : {"auto", None, "slider", "spectrum", Signal}
+            If "auto", if navigation_dimension > 0, a navigator is
+            provided to explore the data.
+            If navigation_dimension is 1 and the signal is an image
+            the navigator is a spectrum obtained by integrating
+            over the signal axes (the image).
+            If navigation_dimension is 1 and the signal is a spectrum
+            the navigator is an image obtained by stacking horizontally
+            all the spectra in the dataset.
+            If navigation_dimension is > 1, the navigator is an image
+            obtained by integrating the data over the signal axes.
+            Additionaly, if navigation_dimension > 2 a window
+            with one slider per axis is raised to navigate the data.
+            For example,
+            if the dataset consists of 3 navigation axes X, Y, Z and one
+            signal axis, E, the default navigator will be an image
+            obtained by integrating the data over E at the current Z
+            index and a window with sliders for the X, Y and Z axes
+            will be raised. Notice that changing the Z-axis index
+            changes the navigator in this case.
+            If "slider" and the navigation dimension > 0 a window
+            with one slider per axis is raised to navigate the data.
+            If "spectrum" and navigation_dimension > 0 the navigator
+            is always a spectrum obtained by integrating the data
+            over all other axes.
+            If None, no navigator will be provided.
+            Alternatively a Signal instance can be provided. The signal
+            dimension must be 1 (for a spectrum navigator) or 2 (for a
+            image navigator) and navigation_shape must be 0 (for a static
+            navigator) or navigation_shape + signal_shape must be equal
+            to the navigator_shape of the current object (for a dynamic
+            navigator).
+            If the signal dtype is RGB or RGBA this parameters has no
+            effect and is always "slider".
+        axes_manager : {None, axes_manager}
+            If None `axes_manager` is used.
+        colorbar : bool, optional
+             If true, a colorbar is plotted for non-RGB images.
+        scalebar : bool, optional
+            If True and the units and scale of the x and y axes are the same a
+            scale bar is plotted.
+        scalebar_color : str, optional
+            A valid MPL color string; will be used as the scalebar color.
+        axes_ticks : {None, bool}, optional
+            If True, plot the axes ticks. If None axes_ticks are only
+            plotted when the scale bar is not plotted. If False the axes ticks
+            are never plotted.
+        auto_contrast : bool, optional
+            If True, the contrast is stretched for each image using the
+            percentile value.
+        percentile : float
```

Comment:
Testing this with the following code gave an error (shown below). It appears that this occurs with any `percentile` value >= 0.5.


```python
im = signals.Image(np.random.random((64, 64)))
im.data[:10, :10] = .2
im.plot(colorbar=True, scalebar=False, axes_ticks=False,auto_contrast=True,percentile=.5)
```

```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-9-9988d6833cd8> in <module>()
      1 im = signals.Image(np.random.random((64, 64)))
      2 im.data[:10, :10] = .2
----> 3 im.plot(colorbar=True, scalebar=False, axes_ticks=False,auto_contrast=True,percentile=.5)

d:\umd_research\software\hyperspy\hyperspy\_signals\image.pyc in plot(self, colorbar, scalebar, scalebar_color, axes_ticks, auto_contrast, percentile, vmin, vmax, no_nans)
    134             vmin=vmin,
    135             vmax=vmax,
--> 136             no_nans=no_nans,
    137         )

d:\umd_research\software\hyperspy\hyperspy\signal.pyc in plot(self, navigator, axes_manager, **kargs)
   3096                     " \"slider\", None, a Signal instance")
   3097 
-> 3098         self._plot.plot(**kargs)
   3099 
   3100     def save(self, filename=None, overwrite=None, extension=None,

d:\umd_research\software\hyperspy\hyperspy\drawing\mpl_he.pyc in plot(self, **kwargs)
    125                 self.pointer.color = 'red'
    126             self.plot_navigator()
--> 127         self.plot_signal(**kwargs)
    128 
    129     def assign_pointer(self):

d:\umd_research\software\hyperspy\hyperspy\drawing\mpl_hie.pyc in plot_signal(self, colorbar, scalebar, scalebar_color, axes_ticks, auto_contrast, percentile, vmin, vmax, no_nans)
     78         imf.scalebar_color = scalebar_color
     79         imf.auto_contrast = auto_contrast
---> 80         imf.plot()
     81         self.signal_plot = imf
     82 

d:\umd_research\software\hyperspy\hyperspy\drawing\image.pyc in plot(self)
    224                 color='red',
    225                 animated=True)
--> 226         self.update()
    227         if self.scalebar is True:
    228             if self.pixel_units is not None:

d:\umd_research\software\hyperspy\hyperspy\drawing\image.pyc in update(self, auto_contrast)
    308                            aspect=self._aspect,
    309                            animated=True)
--> 310             self.figure.canvas.draw()
    311 
    312     def _update(self):

c:\Python279-x64\lib\site-packages\matplotlib\backends\backend_qt5agg.pyc in draw(self)
    146         # causes problems with code that uses the result of the
    147         # draw() to update plot elements.
--> 148         FigureCanvasAgg.draw(self)
    149         self._priv_update()
    150 

c:\Python279-x64\lib\site-packages\matplotlib\backends\backend_agg.pyc in draw(self)
    467 
    468         try:
--> 469             self.figure.draw(self.renderer)
    470         finally:
    471             RendererAgg.lock.release()

c:\Python279-x64\lib\site-packages\matplotlib\artist.pyc in draw_wrapper(artist, renderer, *args, **kwargs)
     57     def draw_wrapper(artist, renderer, *args, **kwargs):
     58         before(artist, renderer)
---> 59         draw(artist, renderer, *args, **kwargs)
     60         after(artist, renderer)
     61 

c:\Python279-x64\lib\site-packages\matplotlib\figure.pyc in draw(self, renderer)
   1077         dsu.sort(key=itemgetter(0))
   1078         for zorder, a, func, args in dsu:
-> 1079             func(*args)
   1080 
   1081         renderer.close_group('figure')

c:\Python279-x64\lib\site-packages\matplotlib\artist.pyc in draw_wrapper(artist, renderer, *args, **kwargs)
     57     def draw_wrapper(artist, renderer, *args, **kwargs):
     58         before(artist, renderer)
---> 59         draw(artist, renderer, *args, **kwargs)
     60         after(artist, renderer)
     61 

c:\Python279-x64\lib\site-packages\matplotlib\axes\_base.pyc in draw(self, renderer, inframe)
   2090 
   2091         for zorder, a in dsu:
-> 2092             a.draw(renderer)
   2093 
   2094         renderer.close_group('axes')

c:\Python279-x64\lib\site-packages\matplotlib\artist.pyc in draw_wrapper(artist, renderer, *args, **kwargs)
     57     def draw_wrapper(artist, renderer, *args, **kwargs):
     58         before(artist, renderer)
---> 59         draw(artist, renderer, *args, **kwargs)
     60         after(artist, renderer)
     61 

c:\Python279-x64\lib\site-packages\matplotlib\image.pyc in draw(self, renderer, *args, **kwargs)
    371                               " correctly with this backend.")
    372 
--> 373             im = self.make_image(renderer.get_image_magnification())
    374             if im is None:
    375                 return

c:\Python279-x64\lib\site-packages\matplotlib\image.pyc in make_image(self, magnification)
    595         im, xmin, ymin, dxintv, dyintv, sx, sy = \
    596             self._get_unsampled_image(self._A, [_x1, _x2, _y1, _y2],
--> 597                                       transformed_viewLim)
    598 
    599         fc = self.axes.patch.get_facecolor()

c:\Python279-x64\lib\site-packages\matplotlib\image.pyc in _get_unsampled_image(self, A, image_extents, viewlim)
    210             else:
    211                 if self._rgbacache is None:
--> 212                     x = self.to_rgba(self._A, bytes=False)
    213                     # Avoid side effects: to_rgba can return its argument
    214                     # unchanged.

c:\Python279-x64\lib\site-packages\matplotlib\cm.pyc in to_rgba(self, x, alpha, bytes)
    260         # This is the normal case, mapping a scalar array:
    261         x = ma.asarray(x)
--> 262         x = self.norm(x)
    263         x = self.cmap(x, alpha=alpha, bytes=bytes)
    264         return x

c:\Python279-x64\lib\site-packages\matplotlib\colors.pyc in __call__(self, value, clip)
    913             result.fill(0)   # Or should it be all masked?  Or 0.5?
    914         elif vmin > vmax:
--> 915             raise ValueError("minvalue must be less than or equal to maxvalue")
    916         else:
    917             vmin = float(vmin)

ValueError: minvalue must be less than or equal to maxvalue
```